### PR TITLE
chore(ctb): `LibPosition` documentation, tests, perf

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds some documentation to `LibPosition` as well as several enhancements suggested by @trianglesphere.

Also adds a new static unit test, `test_pos_correctness`, that asserts the correctness of all gindex, (depth, indexAtDepth), and trace index combinations for a tree with a max depth of 4.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6080** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6080" target="_blank">Graphite</a> 
    * **PR #6081** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6081" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6081?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/dispute/lib/LibPosition.sol [hunk 1]
```diff
@@ -8,20 +8,27 @@ import "../../libraries/DisputeTypes.sol";
  * @notice This library contains helper functions for working with the `Position` type.
  */
 library LibPosition {
+    /**
+     * @notice Computes a generalized index (2^{depth} + indexAtDepth).
+     * @param _depth The depth of the position.
+     * @param _indexAtDepth The index at the depth of the position.
+     * @return position_ The computed generalized index.
+     */
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/lib/LibPosition.sol:16]: Very nice

File: packages/contracts-bedrock/contracts/test/LibPosition.t.sol [hunk 2]
```diff
@@ -144,9 +144,265 @@ contract LibPosition_Test is Test {
...
 
         assertEq(defend.depth(), _depth + 1);
         assertEq(defend.indexAtDepth(), ((_indexAtDepth / 2) * 2 + 1) * 2);
     }
+
+    /**
+     * @notice A static unit test for the correctness of all gindicies, (depth, index) combos,
+     *         and the trace index in a tree of max depth = 4.
+     */
+    function test_pos_correctness() public {
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/LibPosition.t.sol:157]: Style guide asks for specific suffixes, should use either `succeeds` or `works` if you're just testing correctness.
see: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/STYLE_GUIDE.md#contract-naming-conventions
```suggestion
    function test_pos_succeeds() public {
```

**Review by refcell — CHANGES_REQUESTED** [review]

File: packages/contracts-bedrock/contracts/test/LibPosition.t.sol (hunk 2 above)
Comment (clabby) [reply packages/contracts-bedrock/contracts/test/LibPosition.t.sol:157]: Good looks 👍

File: packages/contracts-bedrock/contracts/dispute/lib/LibPosition.sol [hunk 3]
```diff
@@ -91,45 +102,34 @@ library LibPosition {
...
-
-    /**
-     * @notice Get the defense position relative to `position`.
-     * @param _position The position to get the relative defense position of.
-     * @return defense_ The defense position relative to `position`.
+     * @notice Get the move position of `_position`, which is the left child of:
+     *         1. `_position + 1` if `_isAttack` is true.
+     *         1. `_position` if `_isAttack` is false.
+     * @param _position The position to get the relative attack/defense position of.
+     * @param _isAttack Whether or not the move is a defense move.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/lib/LibPosition.sol:127]: This reads better to me, but feel free to dismiss :)
```suggestion
     * @param _isAttack Whether or not the move is an attack move.
```

**Review by refcell — CHANGES_REQUESTED** [review]
Just one last nit :)

**Review by refcell — APPROVED** [review]
LGTM

**Pull request merged**